

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Biology-Specific Features &mdash; Clustergrammer 1.1.0 documentation</title>
  

  
  
    <link rel="shortcut icon" href="_static/graham_cracker.ico"/>
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="https://unpkg.com/font-awesome@4.5.0/css/font-awesome.min.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Matrix Formats and Input/Output" href="matrix_format_io.html" />
    <link rel="prev" title="Interacting with the Visualization" href="interacting_with_viz.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Clustergrammer
          

          
          </a>

          
            
            
              <div class="version">
                1.1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting Started</a></li>
<li class="toctree-l1"><a class="reference internal" href="case_studies.html">Case Studies and Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer2.html">Clustergrammer2</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_gl.html">Clustergrammer-GL</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_web.html">Clustergrammer-Web</a></li>
<li class="toctree-l1"><a class="reference internal" href="interacting_with_viz.html">Interacting with the Visualization</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Biology-Specific Features</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#mouseover-gene-name-and-description">Mouseover Gene Name and Description</a></li>
<li class="toctree-l2"><a class="reference internal" href="#enrichment-analysis">Enrichment Analysis</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="matrix_format_io.html">Matrix Formats and Input/Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="building_webpage.html">Web-Development</a></li>
<li class="toctree-l1"><a class="reference internal" href="app_integration.html">App Integrations</a></li>
<li class="toctree-l1"><a class="reference internal" href="developing_with_clustergrammer.html">Developing</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_widget.html">Clustergrammer-Widget</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_js.html">Clustergrammer-JS</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_py.html">Clustergrammer-PY</a></li>
<li class="toctree-l1"><a class="reference internal" href="license.html">License</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Clustergrammer</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Biology-Specific Features</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/biology_specific_features.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="biology-specific-features">
<span id="id1"></span><h1>Biology-Specific Features<a class="headerlink" href="#biology-specific-features" title="Permalink to this headline">¶</a></h1>
<p>Clustergrammer was developed to visualize high-dimensional biological data (e.g. genome-wide expression data), but it can also generally be applied to any high-dimensional data. Clustergrammer has several biology-specific features that facilitate the analysis of gene-level biological data, such as: gene expression data, proteomics-data, etc. To take advantage of these features, row names must be official gene names. See the <a class="reference external" href="http://amp.pharm.mssm.edu/clustergrammer/CCLE/">CCLE Explorer</a> for examples of visualizing gene expression data. These optional biology-specific features are available in the <a class="reference internal" href="clustergrammer_web.html#clustergrammer-web"><span class="std std-ref">Clustergrammer-Web</span></a> as well as the <a class="reference internal" href="clustergrammer_widget.html#clustergrammer-widget"><span class="std std-ref">Clustergrammer-Widget</span></a> and will automatically activate if the row-names are genes.</p>
<div class="section" id="mouseover-gene-name-and-description">
<span id="hzome-gene-info"></span><h2>Mouseover Gene Name and Description<a class="headerlink" href="#mouseover-gene-name-and-description" title="Permalink to this headline">¶</a></h2>
<p>The human genome consists of over 20,000 genes and modern high-throughput measurements are capable of making measurements across the entire genome (e.g. genome-wide expression studies). Human genes have official gene symbols (e.g. <em>EGFR</em>) that are frequently used to label genes in these datasets. Since no biologist can be knowledgeable about every gene in the genome a common and repetitive task is looking up the names and descriptions of genes in a dataset or visualization. To streamline this activity, Clustergrammer automatically displays the full name and description of a gene (provided by data aggregated through the <a class="reference external" href="http://amp.pharm.mssm.edu/Harmonizome/">Harmonizome</a>) as a tooltip when a user mouses over a gene label (see screenshot below). This simple feature speeds up analysis of large gene-level datasets.</p>
<div class="figure align-left" id="id2">
<a class="reference internal image-reference" href="_images/gene_info.png"><img alt="Gene Info" src="_images/gene_info.png" style="width: 700px;" /></a>
<p class="caption"><span class="caption-text">Mousing over a gene name row brings up the full gene name and description (provided by data aggregated through the <a class="reference external" href="http://amp.pharm.mssm.edu/Harmonizome/">Harmonizome</a>).</span></p>
</div>
<p>The JavaScript file <a class="reference external" href="https://github.com/MaayanLab/clustergrammer/blob/master/js/hzome_functions.js">hzome_functions.js</a> provides this functionality by utilizing <a class="reference external" href="http://amp.pharm.mssm.edu/Harmonizome/documentation">Harmonizome’s RESTful API</a> to obtain gene names and descriptions on mouseover events. <a class="reference external" href="https://github.com/MaayanLab/clustergrammer/blob/master/js/hzome_functions.js">hzome_functions.js</a> is passed to <a class="reference internal" href="clustergrammer_js.html#clustergrammer-js"><span class="std std-ref">Clustergrammer-JS</span></a> as a callback function. See <a class="reference external" href="https://github.com/MaayanLab/clustergrammer/blob/master/js/load_clustergram.js">load_clustergram.js</a> for an example use case. Mouseover callback functions can be used by developers to extend functionality for other domain-specific problems.</p>
</div>
<div class="section" id="enrichment-analysis">
<h2>Enrichment Analysis<a class="headerlink" href="#enrichment-analysis" title="Permalink to this headline">¶</a></h2>
<p>The field of biology has amassed an enormous amount of information about the genes in living organisms such as: function, disease-association, up-stream regulators, protein-level binding partners, etc. Integration of this information can help biologists understand patterns in their data. For instance, enrichment analysis is a popular method to identify biological information specific to a list of genes – e.g. a biologist can use enrichment analysis to identify up-stream regulatory transcription factors that specifically target their a set of up-regulated genes and thereby form hypotheses about potential up-stream regulators.</p>
<p><strong>Export to Enrichr</strong></p>
<p>When a user visualizes a matrix with genes as rows, Clustergrammer automatically enables integration with the enrichment analysis tool <a class="reference external" href="http://amp.pharm.mssm.edu/Enrichr/">Enrichr</a>. Users can export a set of clustered genes to <a class="reference external" href="http://amp.pharm.mssm.edu/Enrichr/">Enrichr</a> using the interactive dendrogram (see screenshot) or import enriched terms into the visualization using the <a class="reference internal" href="#enrichrgram"><span class="std std-ref">Enrichrgram</span></a> functionality.</p>
<div class="figure align-left" id="id3">
<a class="reference internal image-reference" href="_images/send_to_Enrichr_modal.png"><img alt="Send to Enrichr Modal" src="_images/send_to_Enrichr_modal.png" style="width: 600px;" /></a>
<p class="caption"><span class="caption-text">Clicking a row dendrogram cluster opens a modal window with cluster information, row names, and a ‘Send genes to Enrichr’ link that allows users to export their gene list (e.g. cluster of row-genes) to Enrichr.</span></p>
</div>
<p id="enrichrgram"><strong>Enrichrgram</strong></p>
<p>Enrichrgram enables users to find biological information specific to their genes of interest (using <a class="reference external" href="http://amp.pharm.mssm.edu/Enrichr/">Enrichr</a>) and import this information directly into the visualization as row categories (see screenshot below). Enrichrgram can be run on the front or back end (using the <a class="reference internal" href="clustergrammer_py.html#clustergrammer-py-api"><span class="std std-ref">Clustergrammer-PY API</span></a> to pre-calculate results). This feature enables enrichment analysis to be performed within the visualization itself by both the original author of the visualization and subsequent viewers.</p>
<p><strong>Enrichrgram Front End</strong></p>
<p>Enrichrgram on the front end is available to anyone viewing the visualization and can be used by simply clicking the red DNA-like Enrichr logo on the top left of the heatmap, which brings up a list of Enrichr libraries to choose from. To perform enrichment analysis, choose a library and Enrichrgram will return enriched terms from this library that are specifically associated with your list of genes (P-value bars indicate the degree of specificity). For instance, clicking on <em>ChEA 2016</em> will calculate enrichment for up-stream transcription factors. The enriched terms are shown as row categories, which enables users to see which genes are associated with each term. Row-category titles show the enriched term and the red-bars represent the significance of the enrichment (see <a class="reference external" href="http://amp.pharm.mssm.edu/Enrichr/help#basics">Enrichr combined score</a>). Users can run enrichment analysis on a specific cluster of genes by filtering the matrix to only show only their genes of interest. This filtering can be done using the Dendrogram Crop buttons (see <a class="reference internal" href="interacting_with_viz.html#interactive-dendrogram"><span class="std std-ref">Interactive Dendrogram</span></a>) or Brush-Crop button (see <a class="reference internal" href="interacting_with_viz.html#crop"><span class="std std-ref">Cropping</span></a>) to select a subset of genes for analysis.</p>
<div class="figure align-left" id="id4">
<a class="reference internal image-reference" href="_images/enrichrgram_results.png"><img alt="Enrichrgram Menu" src="_images/enrichrgram_results.png" style="width: 900px;" /></a>
<p class="caption"><span class="caption-text">Users can perform enrichment analysis to find biological information specific to their genes (e.g. a cluster of genes). Users can select from several enrichment libraries, and the top 10 enriched terms will be shown as rows categories. The combined scores for the enriched terms will be shown as red bars behind the row category titles.</span></p>
</div>
<p>Note that Enrichrgram results run on the front end are not permanent and will be lost after refreshing the page, but the matrix with enrichment results can be downloaded using the <a class="reference internal" href="interacting_with_viz.html#download"><span class="std std-ref">Download Icon</span></a> button. Enrichment results can be permanently added to the visualization from the back end using the <code class="docutils literal notranslate"><span class="pre">enrichgram</span></code> method described below.</p>
<p><strong>Enrichrgram Back End</strong></p>
<p>To permanently add pre-calculated enrichment results to a visualization run the <code class="docutils literal notranslate"><span class="pre">enrichrgram</span></code> method described in the <a class="reference internal" href="clustergrammer_py.html#clustergrammer-py-api"><span class="std std-ref">Clustergrammer-PY API</span></a> before clustering. The Jupyter notebook <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/CCLE_Clustergrammer/blob/master/notebooks/Clustergrammer_CCLE_Notebook.ipynb">Clustergrammer_CCLE_Notebook.ipynb</a> demonstrates how to use the <code class="docutils literal notranslate"><span class="pre">enrichrgram</span></code> method to pre-calculate enrichment analysis results for a visualization.</p>
<p>The <a class="reference external" href="https://github.com/MaayanLab/clustergrammer/blob/master/js/Enrichrgram.js">Enrichrgram.js</a> library provides this functionality on the front end and works with the <a class="reference internal" href="clustergrammer_js.html#clustergrammer-js"><span class="std std-ref">Clustergrammer-JS</span></a> API to depict enriched terms and their associated genes as row categories. The update-row-category functionality can be extended by developers for other domain-specific problems.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="matrix_format_io.html" class="btn btn-neutral float-right" title="Matrix Formats and Input/Output" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="interacting_with_viz.html" class="btn btn-neutral" title="Interacting with the Visualization" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Nicolas F. Fernandez

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    
    
      <script type="text/javascript">
          var DOCUMENTATION_OPTIONS = {
              URL_ROOT:'./',
              VERSION:'1.1.0',
              LANGUAGE:'None',
              COLLAPSE_INDEX:false,
              FILE_SUFFIX:'.html',
              HAS_SOURCE:  true,
              SOURCELINK_SUFFIX: '.txt'
          };
      </script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script type="text/javascript" src="https://unpkg.com/@jupyter-widgets/html-manager@^0.18.0/dist/embed-amd.js"></script>
    

  

  <script type="text/javascript" src="_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>