

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Case Studies and Tutorials &mdash; Clustergrammer 1.1.0 documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="https://unpkg.com/font-awesome@4.5.0/css/font-awesome.min.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Clustergrammer-Web" href="clustergrammer_web.html" />
    <link rel="prev" title="Getting Started" href="getting_started.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Clustergrammer
          

          
          </a>

          
            
            
              <div class="version">
                1.1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting Started</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Case Studies and Tutorials</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#single-cell-gene-expression-2-700-pbmc">Single Cell Gene Expression 2,700 PBMC</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#view-interactive-notebook">View Interactive Notebook</a></li>
<li class="toctree-l3"><a class="reference internal" href="#run-notebook-on-the-cloud">Run Notebook on the Cloud</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#cancer-cell-line-encyclopedia-gene-expression-data">Cancer Cell Line Encyclopedia Gene Expression Data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#lung-cancer-post-translational-modification-and-gene-expression-regulation">Lung Cancer Post-Translational Modification and Gene Expression Regulation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#single-cell-cytof-dataset">Single Cell CyTOF Dataset</a></li>
<li class="toctree-l2"><a class="reference internal" href="#large-network-kinase-substrate-similarity-network">Large Network: Kinase Substrate Similarity Network</a></li>
<li class="toctree-l2"><a class="reference internal" href="#machine-learning-and-miscellaneous-datasets">Machine Learning and Miscellaneous Datasets</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_web.html">Clustergrammer-Web</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer2.html">Clustergrammer2 Widget</a></li>
<li class="toctree-l1"><a class="reference internal" href="interacting_with_viz.html">Interacting with the Visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="biology_specific_features.html">Biology-Specific Features</a></li>
<li class="toctree-l1"><a class="reference internal" href="matrix_format_io.html">Matrix Formats and Input/Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="building_webpage.html">Web-Development with Clustergrammer</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_widget.html">Clustergrammer-Widget</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_js.html">Clustergrammer-JS</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_gl.html">Clustergrammer-GL</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustergrammer_py.html">Clustergrammer-PY</a></li>
<li class="toctree-l1"><a class="reference internal" href="app_integration.html">App Integration Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="developing_with_clustergrammer.html">Developing Clustergrammer</a></li>
<li class="toctree-l1"><a class="reference internal" href="license.html">License</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Clustergrammer</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Case Studies and Tutorials</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/case_studies.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="case-studies-and-tutorials">
<span id="case-studies"></span><h1>Case Studies and Tutorials<a class="headerlink" href="#case-studies-and-tutorials" title="Permalink to this headline">¶</a></h1>
<p>Clustergrammer was developed to visualize high-dimensional biological data (e.g. genome-wide expression data), but it can also generally be applied to any high-dimensional data. Below are links to several case studies and examples using Clustergrammer to explore high-dimensional data. All examples are below are publically available through GitHub.</p>
<div class="section" id="single-cell-gene-expression-2-700-pbmc">
<h2>Single Cell Gene Expression 2,700 PBMC<a class="headerlink" href="#single-cell-gene-expression-2-700-pbmc" title="Permalink to this headline">¶</a></h2>
<div style="position: relative; padding-bottom: 10px; height: 0; overflow: hidden; max-width: 100%; height: auto;">

<iframe width="560" height="315" src="https://www.youtube.com/embed/f-snXe2Bn9Q" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
</div><p>Single cell RNA-seq (scRNA-seq) is a powerful method to interrogate gene expression across thousands of single cells. This method provides thousands of measurements (single cells) across thousands of dimensions (genes). Clustergrammer2 is used to interactively explore an example dataset of 2,700 PBMCs obtained from <a class="reference external" href="https://www.10xgenomics.com/resources/datasets/">10X Genomics</a>. Bulk gene expression signatures of cell types obtained from <a class="reference external" href="https://cibersort.stanford.edu/">CIBERSORT</a> were used to obtain a tentative cell type for each cell. The data and code can be found on GitHub at <a class="reference external" href="https://github.com/ismms-himc/clustergrammer2-notebooks">clustergrammer2-notebooks</a> and the notebook can be viewed and re-run on the cloud - see below.</p>
<div class="section" id="view-interactive-notebook">
<h3>View Interactive Notebook<a class="headerlink" href="#view-interactive-notebook" title="Permalink to this headline">¶</a></h3>
<p>The interactive pre-run notebook can be viewed on NBVIewer.</p>
<ul class="simple">
<li>NBViewer: <a class="reference external" href="https://nbviewer.jupyter.org/github/ismms-himc/clustergrammer2-notebooks/blob/master/notebooks/3.0_10X_Genomics_2700_Dataset_NBViewer.ipynb">NBViewer-2700</a>.</li>
</ul>
</div>
<div class="section" id="run-notebook-on-the-cloud">
<h3>Run Notebook on the Cloud<a class="headerlink" href="#run-notebook-on-the-cloud" title="Permalink to this headline">¶</a></h3>
<p>This notebook can be re-run and modified on the cloud using:</p>
<ul class="simple">
<li>MyBinder: <a class="reference external" href="https://mybinder.org/v2/gh/ismms-himc/clustergrammer2_examples/master?filepath=notebooks%2F3.0_10X_Genomics_2700_Dataset.ipynb">MyBinder-2700</a></li>
<li>Kaggle: <a class="reference external" href="https://www.kaggle.com/cornhundred/viz-2-700-pbmcs-with-cell-type-predictions">Viz-2-700-pbmcs-with-cell-type-predictions</a>, <a class="reference external" href="https://www.kaggle.com/cornhundred/clustergrammer2-2-700-pbmc-notebook">Clustergrammer2-2-700-pbmc-notebook</a></li>
<li>Saturn Cloud: <a class="reference external" href="https://www.saturncloud.io/published/cornhundred/clustergrammer2-notebooks/notebooks/pbmc_2700_scRNA-seq/1.0_PBMC_2700_scRNA-seq.ipynb">Saturn-2700</a></li>
</ul>
</div>
</div>
<div class="section" id="cancer-cell-line-encyclopedia-gene-expression-data">
<h2>Cancer Cell Line Encyclopedia Gene Expression Data<a class="headerlink" href="#cancer-cell-line-encyclopedia-gene-expression-data" title="Permalink to this headline">¶</a></h2>
<div class="figure align-left" id="id1">
<a class="reference external image-reference" href="https://maayanlab.github.io/CCLE_Clustergrammer/"><img alt="CCLE Explorer" src="_images/CCLE_explorer.png" style="width: 800px;" /></a>
<p class="caption"><span class="caption-text">Screenshot from the <a class="reference external" href="http://amp.pharm.mssm.edu/clustergrammer/CCLE/">CCLE Explorer</a> showing the tissue breakdown of the CCLE. Clicking on a tissue brings up an interactive heatmap with the top 250 most variable genes within a tissue. Also see the <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/CCLE_Clustergrammer/blob/master/notebooks/Clustergrammer_CCLE_Notebook.ipynb">CCLE Jupyter Notebook</a> for an example of how to explore the CCLE gene expression data in a Jupyter notebook.</span></p>
</div>
<p>The Cancer Cell Line Encyclopedia (<a class="reference external" href="https://portals.broadinstitute.org/ccle/home">CCLE</a>) is a publicly available project that has characterized (e.g. genetic characterization) over 1,000 cancer cell lines. We used Clustergrammer to re-analyze and visualize CCLE’s gene expression data in the <a class="reference external" href="http://amp.pharm.mssm.edu/clustergrammer/CCLE/">CCLE Explorer</a>. The CCLE Explorer allows users to explore the CCLE by tissue type and visualize the most commonly differentially expressed genes for each tissue type as an interactive heatmap. The <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/CCLE_Clustergrammer/blob/master/notebooks/Clustergrammer_CCLE_Notebook.ipynb">CCLE Jupyter Notebook</a> generates an overview of the CCLE gene expression data, investigates specific tissues, and explains how to use <a class="reference internal" href="biology_specific_features.html#enrichrgram"><span class="std std-ref">Enrichrgram</span></a> to understand the biological functions of differentially expressed genes.</p>
</div>
<div class="section" id="lung-cancer-post-translational-modification-and-gene-expression-regulation">
<h2>Lung Cancer Post-Translational Modification and Gene Expression Regulation<a class="headerlink" href="#lung-cancer-post-translational-modification-and-gene-expression-regulation" title="Permalink to this headline">¶</a></h2>
<div class="figure align-left" id="id2">
<a class="reference external image-reference" href="http://nbviewer.jupyter.org/github/MaayanLab/CST_Lung_Cancer_Viz/blob/master/notebooks/CST_Data_Viz.ipynb"><img alt="CST Screenshot" src="_images/CST_screenshot.png" style="width: 450px;" /></a>
<p class="caption"><span class="caption-text">Screenshot from the <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/CST_Lung_Cancer_Viz/blob/master/notebooks/CST_Data_Viz.ipynb?flush_cache=true">CST_Data_Viz.ipynb</a> Jupyter notebook showing hierarchical clustering of differential phosphorylation, methylation, acetylation, and gene expression data across 37 lung cancer cell lines. See the interactive Jupyter notebook <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/CST_Lung_Cancer_Viz/blob/master/notebooks/CST_Data_Viz.ipynb?flush_cache=true">CST_Data_Viz.ipynb</a> for more information.</span></p>
</div>
<p>Lung cancer is a complex disease that is known to be regulated at the post-translational modification (PTM) level, e.g. phosphorylation driven by kinases. Our collaborators at <a class="reference external" href="https://www.cellsignal.com/">Cell Signaling Technology Inc</a> used Tandem Mass Tag (TMT) mass spectrometry to measure differential phosphorylation, acetylation, and methylation in a panel of 42 lung cancer cell lines compared to non-cancerous lung tissue. Gene expression data from 37 of these lung cancer cell lines was also independently obtained from the publicly available Cancer Cell Line Encyclopedia (<a class="reference external" href="https://portals.broadinstitute.org/ccle/home">CCLE</a>). In the Jupyter notebook <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/CST_Lung_Cancer_Viz/blob/master/notebooks/CST_Data_Viz.ipynb?flush_cache=true">CST_Data_Viz.ipynb</a> we:</p>
<ul class="simple">
<li>Visualize PTM data, gene expression data, and merged PTM/gene-expression data</li>
<li>Identify co-regulated clusters of PTMs/genes in distinct lung cancer cell line subtypes</li>
<li>Perform enrichment analysis to understand the biological processes involved in PTM/expression clusters</li>
</ul>
</div>
<div class="section" id="single-cell-cytof-dataset">
<h2>Single Cell CyTOF Dataset<a class="headerlink" href="#single-cell-cytof-dataset" title="Permalink to this headline">¶</a></h2>
<div class="figure align-left" id="id3">
<a class="reference external image-reference" href="http://nbviewer.jupyter.org/github/MaayanLab/Cytof_Plasma_PMA/blob/master/notebooks/Plasma_vs_PMA_Phosphorylation.ipynb"><img alt="CyTOF Screenshot" src="_images/CyTOF_screenshot.png" style="width: 450px;" /></a>
<p class="caption"><span class="caption-text">Screenshot from the <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/Cytof_Plasma_PMA/blob/master/notebooks/Plasma_vs_PMA_Phosphorylation.ipynb">Plasma_vs_PMA_Phosphrylation.ipynb</a> Jupyter notebook showing downsampled single cell CyTOF data (K-means downsampled from 220,000 single cells to 2,000 cell-clusters). Cell-clusters are shown as rows with cell-type categories (e.g. Natural Killer cells) and phosphorylations are shown as columns. See the interactive Jupyter notebook <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/Cytof_Plasma_PMA/blob/master/notebooks/Plasma_vs_PMA_Phosphorylation.ipynb">Plasma_vs_PMA_Phosphrylation.ipynb</a> for more information.</span></p>
</div>
<p>Our collaborators in the <a class="reference external" href="http://labs.icahn.mssm.edu/giannarellilab/">Giannarelli Lab</a> and the <a class="reference external" href="http://icahn.mssm.edu/research/portal/resources/deans-cores/human-immune-monitoring-core">Icahn School of Medicine Human Immune Monitoring Core</a> used Mass Cytometry, CyTOF (Fluidigm), to investigate the phosphorylation response of peripheral blood mononuclear cells (PBMC) immune cells exposed to PMA (phorbol 12-myristate 13-acetate), a tumor promoter and activator of protein kinase C (PKC). A total of 28 markers (18 surface markers and 10 phosphorylation markers) were measured in over 200,000 single cells. In the Jupyter notebook <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/Cytof_Plasma_PMA/blob/master/notebooks/Plasma_vs_PMA_Phosphorylation.ipynb">Plasma_vs_PMA_Phosphrylation.ipynb</a> we semi-automatically identify cell types using surface markers and cluster cells based on phosphorylation to identify cell-type specific behavior at the phosphorylation level. See the <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/Cytof_Plasma_PMA/blob/master/notebooks/Plasma_vs_PMA_Phosphorylation.ipynb">Plasma_vs_PMA_Phosphrylation.ipynb</a> Jupyter notebook for more information.</p>
</div>
<div class="section" id="large-network-kinase-substrate-similarity-network">
<h2>Large Network: Kinase Substrate Similarity Network<a class="headerlink" href="#large-network-kinase-substrate-similarity-network" title="Permalink to this headline">¶</a></h2>
<div class="figure align-left" id="id4">
<a class="reference external image-reference" href="https://maayanlab.github.io/kinase_substrate_similarity_network/"><img alt="Kinase Network Screenshot" src="_images/kinase_network_screenshot.png" style="width: 450px;" /></a>
<p class="caption"><span class="caption-text">Screenshot from the <a class="reference external" href="https://maayanlab.github.io/kinase_substrate_similarity_network/">Kinase Substrate Similarity Network</a> example that demonstrates how Clustergrammer can be used to visualize a large network of kinases based on shared substrates.</span></p>
</div>
<p>Clustergrammer can be used to visualize large networks without the formation of ‘hairballs’. In the <a class="reference external" href="https://maayanlab.github.io/kinase_substrate_similarity_network/">Kinase Substrate Similarity Network</a> example we use Clustergrammer to visualize a network kinases based on shared substrate that includes 404 kinases and 163,216 links. Kinases are shown as rows and columns. For more information see the <a class="reference external" href="https://maayanlab.github.io/kinase_substrate_similarity_network/">Kinase Substrate Similarity Network</a> example.</p>
</div>
<div class="section" id="machine-learning-and-miscellaneous-datasets">
<h2>Machine Learning and Miscellaneous Datasets<a class="headerlink" href="#machine-learning-and-miscellaneous-datasets" title="Permalink to this headline">¶</a></h2>
<div class="figure align-left" id="id5">
<a class="reference external image-reference" href="http://nbviewer.jupyter.org/github/MaayanLab/MNIST_heatmaps/blob/master/notebooks/MNIST_Notebook.ipynb#Visualize-Downsampled-Version-of-MNIST"><img alt="MNIST Screenshot" src="_images/MNIST_screenshot.png" style="width: 450px;" /></a>
<p class="caption"><span class="caption-text">Screenshot from the <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/MNIST_heatmaps/blob/master/notebooks/MNIST_Notebook.ipynb">MNIST Notebook</a> that demonstrates how the <a class="reference internal" href="clustergrammer_widget.html#clustergrammer-widget"><span class="std std-ref">Clustergrammer-Widget</span></a> can be used to visualize the <a class="reference external" href="http://yann.lecun.com/exdb/mnist/">MNIST Data</a>. Downsampled handwritten digits (K-means downsampled from 70,0000 handwritten digits to 300 digit-clusters) are shown as columns with digit-type categories and pixels are shown as rows. For more information see the <a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/MNIST_heatmaps/blob/master/notebooks/MNIST_Notebook.ipynb">MNIST Notebook</a>.</span></p>
</div>
<p>Clustergrammer was used to visualize several widely used machine learning Datasets and other miscellaneous Datasets:</p>
<ul class="simple">
<li><a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/MNIST_heatmaps/blob/master/notebooks/MNIST_Notebook.ipynb">MNIST Handwritten Digit Dataset</a></li>
<li><a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/iris_clustergrammer_visualization/blob/master/Iris%20Dataset.ipynb">Iris Flower Dataset</a></li>
<li><a class="reference external" href="http://nbviewer.jupyter.org/github/MaayanLab/USDA_Nutrients_Viz/blob/master/USDA_Nutrients.ipynb">USDA Nutrient Dataset</a></li>
</ul>
<p>These examples demonstrate the generality of heatmap visualizations and enable users to interactively explore familiar Datasets.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="clustergrammer_web.html" class="btn btn-neutral float-right" title="Clustergrammer-Web" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="getting_started.html" class="btn btn-neutral" title="Getting Started" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, Nicolas F. Fernandez

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    
    
      <script type="text/javascript">
          var DOCUMENTATION_OPTIONS = {
              URL_ROOT:'./',
              VERSION:'1.1.0',
              LANGUAGE:'None',
              COLLAPSE_INDEX:false,
              FILE_SUFFIX:'.html',
              HAS_SOURCE:  true,
              SOURCELINK_SUFFIX: '.txt'
          };
      </script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script type="text/javascript" src="https://unpkg.com/@jupyter-widgets/html-manager@^0.14.0/dist/embed-amd.js"></script>
    

  

  <script type="text/javascript" src="_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>